### PR TITLE
[New Rule] Google Workspace Admin Role Deletion

### LINKED ISSUE #511: [New Rule] Google Workspace Admin Role Deletion
## Description
Detects when a custom admin role is deleted. An adversary may delete a custom admin role in order to impact the permissions or capabilities of system administrators.

## Required Info

- **Eventing Sources:**
`filebeat-*`

- **Platforms**
`Google Workspace (formerly G Suite)`

- **Target ECS Version:** `1.6.0`
- **New fields required in ECS for this?** No
- **Related issues or PRs** N/A

## Optional Info
- **References:**
  - https://support.google.com/a/answer/2406043?hl=en

## Example Logic
```
event.dataset:gsuite.admin and event.provider:admin and event.category:iam and event.action:DELETE_ROLE
```

## Screenshot
image

### PR DESCRIPTION
**Pull request by threat-punter**
## Issues
Resolves #511 

## Summary

Detects when a custom admin role is deleted. An adversary may delete a custom admin role in order to impact the permissions or capabilities of system administrators.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Review by Samirbous — APPROVED** [review]

File: rules/google-workspace/google_workspace_admin_role_deletion.toml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+[metadata]
+creation_date = "2020/11/17"
+ecs_version = ["1.6.0"]
+maturity = "production"
+updated_date = "2020/11/17"
+
+[rule]
+author = ["Elastic"]
+description = """
+Detects when a custom admin role is deleted. An adversary may delete a custom admin role in order to impact the
```
Comment (bm11100) [inline rules/google-workspace/google_workspace_admin_role_deletion.toml:10]: I usually stay away from shortening words like admin, or legit, but in this case I think its fine.

**Review by bm11100 — APPROVED** [review]
🌵

File: rules/google-workspace/google_workspace_admin_role_deletion.toml (hunk 1 above)
Comment (threat-punter) [inline rules/google-workspace/google_workspace_admin_role_deletion.toml:10]:
```suggestion
Detects when a custom administrator role is deleted. An adversary may delete a custom administrator role in order to
impact the permissions or capabilities of system administrators.
```

Comment (threat-punter) [reply rules/google-workspace/google_workspace_admin_role_deletion.toml:10]: I used the shortened terms, as that's what Google uses in their documentation and UI. If you're happy with it, I'll go ahead and merge them 👍

Comment (threat-punter) [reply rules/google-workspace/google_workspace_admin_role_deletion.toml:10]: If Brent is happy, I am happy

**Pull request merged**